Created add-post page

wang-joseph opened the pull request:
Closes #9.

This page will allow users to add posts to the board. A form will be provided for the user to submit to an API endpoint to save a post. This form will be accessible on both mobile and desktop.

To-do list partially adapted from #9 :
- [x] Create foundation for the add-post page
- [x] Create a form asking for name, message, and a photo.
- [x] Error validation if the user didn't fill out both the message + photo field
- [ ] Make the form responsive on mobile site
- [ ] Redirect to home page after adding it
- [ ] Send request to correct API endpoint and request the new post from the server 

Once the backend connection is complete, this page can send request to the correct API endpoint.

wang-joseph committed:
Created basic add-post foundation

Created the basic page and structure to add a page, minus the form elements.

wang-joseph commented:
I do a few interesting things to make sure that this page can resize because the header height is a static px so I want to make sure that they're okay or if we should make something akin to a `settings.json` or if a better solution exists that I haven't thought of.

wang-joseph committed:
adjusted add-post documentation

shari09 commented on src/pages/AddPost.tsx:
```diff
@@ -1,27 +1,73 @@
 export const AddPost: React.FC = () => {
+  const board = useContext<IBoardContext>(BoardContext);
+
+  const [windowHeight, setWindowHeight] = useState<number>(window.innerHeight)
+
+  /**
+   * Changes this add post page's height.
+   */
+  const adjustHeight = () => {
+    setWindowHeight(window.innerHeight)
+  }
+
+  // TODO: figure out more efficient way of making sure the main body is the
+  // right height.
+  // Account for window resizing so that the page and image eight is correct
+  useEffect(() => {
+    window.addEventListener('resize', adjustHeight)
+
+    return () => {
+      window.removeEventListener('resize', adjustHeight)
+    }
+  }, [])
 
   const wrapperStyle: SxStyleProp = {
     variant: 'bodyWrapper',
+    overflow: 'hidden',
+    display: 'flex',
+    flexDirection: 'row',
+
+    // Since the header height is a constant px amount, height should be the
+    // height of the window minus constant
+    height: windowHeight - HEADER_HEIGHT_PX,
   };
 
+  const photoStyle: SxStyleProp = {
+    width: '30%',
+    height: '100%',
+    backgroundImage: `linear-gradient(#EDEDED99, #EDEDED99), url(${board.bgImage})`,
+    backgroundSize: 'cover',
+  }
```
Rather than using `variant: bodyWrapper` which has a mt, change it to a pt and set the `minHeight` to `100vh`. Then, rather than specifying a height for the photo, make it auto so it'll span the entire height.

shari09 reviewed (changes requested):
css adjustments

wang-joseph committed:
Merge branch 'master' into add-on-page

wang-joseph committed:
Created foundation for form

Changed up the board context as the form needs to know the board UUID, which should be stored in the board context as it is directly related to the board

wang-joseph committed:
changed the photo height loading to be more efficient

no need to mount a listener
